cli/submit.go

```diff
@@ -1,96 +1,170 @@
 package cli
 
 import (
-    "bufio"
-    "fmt"
-    "github.com/DataDrake/cli-ng/cmd"
-    "github.com/arkenproject/ait/utils"
-    "github.com/go-git/go-git/v5"
-    "github.com/go-git/go-git/v5/plumbing/object"
-    "log"
-    "os"
-    "path/filepath"
-    "time"
+	"bufio"
+	"fmt"
+	"github.com/DataDrake/cli-ng/cmd"
+	"github.com/arkenproject/ait/keysets"
+	"github.com/arkenproject/ait/utils"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"golang.org/x/crypto/ssh/terminal"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"time"
 )
 
 var Submit = cmd.CMD{
-    Name:  "submit",
-    Short: "Submit your Keyset to a git repository.",
-    Args:  &SubmitArgs{},
-    Run:   SubmitRun,
+	Name:  "submit",
+	Short: "Submit your Keyset to a git repository.",
+	Args:  &SubmitArgs{},
+	Run:   SubmitRun,
 }
 
 type SubmitArgs struct {
-    Args []string
+	Args []string
 }
 
+//A place to put special repo "aliases"
+var SpecialRepos = map[string]string{
+	"core": "https://github.com/arkenproject/core-keyset.git",
+}
+
+//SubmitRun generates a keyset file and then clones the Github repo at the given
+//url, adds the keyset file, commits it, and pushes it, and then deletes the repo
+//once everything is done or if anything goes wrong before completion. With all
+//of those steps, there are MANY possible points of failure. If anything goes
+//wrong, the error will be log.Fatal'd and the repo will we deleted from
+//its temporary location at .ait/sources. Users are not meant to deal with the
+//repos directly at any point so it and the keyset file are basically ephemeral
+//and only exist on disk while this command is running.
 func SubmitRun(_ *cmd.RootCMD, c *cmd.CMD) {
-    args := c.Args.(*SubmitArgs).Args
-    if len(args) < 2 {
-        log.Fatal("Not enough arguments, expected repository name and path to Keyset")
-    }
-    repoName, keysetPath := args[0], args[1]
-    repo := add(repoName, keysetPath)
-    commit(repo)
-    push(repo)
+	args := c.Args.(*SubmitArgs).Args
+	if len(args) < 1 {
+		log.Fatal("Not enough arguments, expected repository url")
+	}
```

Comment:
Have we tested if `cli-ng` won't take care of this automatically?

Reply:
I have not, but i will